packages/next/src/build/swc/options.ts

```diff
@@ -17,7 +17,11 @@ const regeneratorRuntimePath = require.resolve(
 )
 
 function isTypeScriptFile(filename: string) {
-  return filename.endsWith('.ts') || filename.endsWith('.tsx')
+  return (
+    filename.endsWith('.ts') ||
+    filename.endsWith('.mts') ||
```

Comment:
To allow `next-swc-loader` to parse `.mts` extensions such as `next.config.mts`.